### PR TITLE
feat: Shopping cart and edit cart

### LINKED ISSUE #22: Edit Cart
- Add button onto product view and store view
- modify quantity

### PR DESCRIPTION
**Pull request by dershao**
Adds a shopping cart functionality in the code.

Contains:
- Allows users to add to a shopping cart
- Shows items of the shopping cart in a shopping cart view
- Edit the quantities and remove items on the shopping cart view

It does not have:
- ability to save cart state with cookie/session

shoppingcart

Closes #22 #21

### CODE REVIEW

File: shopify-client/src/components/Header.vue [hunk 1]
```diff
@@ -25,6 +25,7 @@
           <b-nav-item to="/app/about">About</b-nav-item>
           <b-nav-item to="/app/login">Login</b-nav-item>
           <b-nav-item to="/app/signup">Sign Up</b-nav-item>
+          <b-nav-item to="/app/cart">Cart ({{ cartItems }})</b-nav-item>
```
Comment (dershao) [inline shopify-client/src/components/Header.vue:28]: header-cart

**Review by mpfan — CHANGES_REQUESTED** [review]
I know the PR stated that session/cookie is not implemented but I think it should be part of this feature. The implementation of the checkout will be dependent on how the cart tracking is done in this PR. This means we can't easily switch to session/cookie if we chose to use vuex to track the cart. The limitation with vuex is that state is lost upon refresh.

**Comment by dershao** [comment]
> I know the PR stated that session/cookie is not implemented but I think it should be part of this feature. The implementation of the checkout will be dependent on how the cart tracking is done in this PR. This means we can't easily switch to session/cookie if we chose to use vuex to track the cart. The limitation with vuex is that state is lost upon refresh.

From my understanding of cookies, we can't store an object into the cookie. However, an object is useful for this shopping cart feature especially for counting the number of items of each item in the cart. So either way, I feel like some "cart object" will be present. The implementation I was going for was we would just serialize the cart object into a cookie so when the user comes back, it will check the cookie for a cart and deserialize it back into the vuex state.

Therefore, the roadmap for this feature would be something like this: 
- Use vuex to keep track of shopping cart state (no persistence)
- Add cookies support - cart object would be serialized to be stored in a cookie
- When user authentication is added, we would store the cart object into a session on our server and use some kind of session id in the cookie to retrieve the cart object we have stored in the database

Given this implementation, I figured I pushed this code now since for the checkout page, it would also probably use the vuex state to check for the cart object. It's just that in the future, there will be an additional step (e.g., if someone goes to the checkout page directly) it will check for a cart object in the cookies and deserialize that object into the vuex state.

**Review by Souheil-Yazji — APPROVED** [review]
LGTM

File: shopify-client/src/views/Cart.vue [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        <b-row>
+          <b-col class="col-sm-2"></b-col>
+          <b-col>
+            <p>Product Name</p>
+          </b-col>
+          <b-col>
+            <p>Price</p>
+          </b-col>
+        </b-row>
+        <b-row v-for="product in products" :key="product.name" class="b-row-cart-item">
```
Comment (Weebian) [inline shopify-client/src/views/Cart.vue:18]: nit-pick: should be on new lines:

:key=""
class=""

etc.

**Review by Weebian — APPROVED** [review]
+1

**Review by mpfan — APPROVED** [review]

**Pull request merged**